subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformCachingIntegrationTest.groovy

```diff
@@ -1239,6 +1187,9 @@ class ArtifactTransformCachingIntegrationTest extends AbstractHttpDependencyReso
                 task jar2(type: Jar) {
                     archiveName = 'lib2.jar'
                 }
+                tasks.withType(Jar) {
+                    destinationDir = buildDir
```

Comment:
Shouldn't we have some validation for this? Using the current working dir is always gonna be nonsensical.

Reply:
I was thinking about emitting a deprecation warning when `destinationDir` is not set. However, it seems that's really an edge case that only happens when the base plugin is not applied. Thus, I think it may not be worth it. WDYT?

https://github.com/gradle/gradle/blob/b48b84bc4b6ff505c57cee3c9346231f9b60a27f/subprojects/plugins/src/main/java/org/gradle/api/plugins/BasePlugin.java#L106

Reply:
Alright, let's do that. I'll open a separate issue/PR.